src/pages/methods/WorkflowList.test.tsx

```diff
@@ -226,26 +239,108 @@ describe('workflows table', () => {
 
     // Assert
-    expect(table).toHaveAttribute('aria-colcount', '4');
+    expect(table).toHaveAttribute('aria-colcount', '5');
     expect(table).toHaveAttribute('aria-rowcount', '2');
 
     const headers: HTMLElement[] = within(table).getAllByRole('columnheader');
-    expect(headers).toHaveLength(4);
-    expect(headers[0]).toHaveTextContent('Method');
-    expect(headers[1]).toHaveTextContent('Synopsis');
-    expect(headers[2]).toHaveTextContent('Owners');
-    expect(headers[3]).toHaveTextContent('Snapshots');
+    expect(headers).toHaveLength(5);
+    expect(headers[0]).toHaveTextContent('Actions');
+    expect(headers[1]).toHaveTextContent('Method');
+    expect(headers[2]).toHaveTextContent('Synopsis');
+    expect(headers[3]).toHaveTextContent('Owners');
+    expect(headers[4]).toHaveTextContent('Snapshots');
+
+    const rows: HTMLElement[] = within(table).getAllByRole('row');
+    expect(rows).toHaveLength(2);
+
+    const methodCells: HTMLElement[] = within(rows[1]).getAllByRole('cell');
+    expect(methodCells).toHaveLength(5);
+    within(methodCells[1]).getByText('revali bird namespace');
+    within(methodCells[1]).getByText('revali method 2');
+    expect(methodCells[2]).toHaveTextContent('another revali description');
+    expect(methodCells[3]).toHaveTextContent('revali@gale.com, revali@champions.com');
+    expect(methodCells[4]).toHaveTextContent('1');
+  });
+
+  it('allows editing permissions for namespace owned by user', async () => {
+    // Arrange
+    asMockedFn(Methods).mockReturnValue(mockMethods([revaliMethod]));
+    const user: UserEvent = userEvent.setup();
+
+    // set the user's email
+    jest.spyOn(userStore, 'get').mockImplementation(jest.fn().mockReturnValue(mockUserState('revali@gale.com')));
+
+    // Act
+    await act(async () => {
+      render(<WorkflowList />);
+    });
+
+    const table: HTMLElement = await screen.findByRole('table');
 
     const rows: HTMLElement[] = within(table).getAllByRole('row');
     expect(rows).toHaveLength(2);
 
     const methodCells: HTMLElement[] = within(rows[1]).getAllByRole('cell');
-    expect(methodCells).toHaveLength(4);
-    within(methodCells[0]).getByText('revali bird namespace');
-    within(methodCells[0]).getByText('revali method 2');
-    expect(methodCells[1]).toHaveTextContent('another revali description');
-    expect(methodCells[2]).toHaveTextContent('revali@gale.com, revali@champions.com');
-    expect(methodCells[3]).toHaveTextContent('1');
+    expect(methodCells).toHaveLength(5);
+    within(methodCells[1]).getByText('revali bird namespace');
+    within(methodCells[1]).getByText('revali method');
+
+    const actionsMenu = within(methodCells[0]).getByRole('button');
+
+    // Act
+    await user.click(actionsMenu);
+
+    // Assert
+    expect(screen.getByText('Edit namespace permissions'));
+
+    // Act
+    await user.click(screen.getByRole('button', { name: 'Edit namespace permissions' }));
+
+    // Assert
+    expect(
```

Comment:
It wouldn't hurt to add something after this that checks that the workflow list actually uses the namespace permissions provider and not the snapshot one for its permissions modal.

Reply:
I am not sure what would that test prove? Because in the code we actually set it to namespace vs snapshot one right? And the check to ensure right methods were called I think is already covered in PermissionsModal and PermissionsProvider test

Reply:
Yeah, it would be more of a regression unit test. I suggested it because none of the `WorkflowList` tests make sure the right API call is ultimately made for this particular permissions modal, but we don't need it if you don't think it's necessary